server/stream.go

```diff
@@ -5513,15 +5516,30 @@ func (mset *stream) hasAllPreAcks(seq uint64, subj string) bool {
 // Check if we have all consumers pre-acked.
 // Write lock should be held.
 func (mset *stream) clearAllPreAcks(seq uint64) {
-	delete(mset.preAcks, seq)
+	for o, ss := range mset.preAcks {
+		if ss.Delete(seq) && ss.Size() == 0 {
+			delete(mset.preAcks, o)
+		}
+	}
 }
 
 // Clear all preAcks below floor.
 // Write lock should be held.
 func (mset *stream) clearAllPreAcksBelowFloor(floor uint64) {
-	for seq := range mset.preAcks {
-		if seq < floor {
-			delete(mset.preAcks, seq)
+	for o, ss := range mset.preAcks {
+		min, max := ss.MinMax()
+		if max < floor {
+			delete(mset.preAcks, o)
+			continue
+		}
+		if min > floor {
```

Comment:
This could be eliminated and just let the for loop on line 5538 take care of it. But with this and 5538 we do not cover min == floor, so something to resolve.

Reply:
The check before was `seq < floor` so this check might want fixing but 5538 should be fine?